crowbar_framework/app/models/updater_service.rb

```diff
@@ -75,6 +105,30 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
     @logger.debug("Updater apply_role_post_chef_call: exiting")
   end
 
+  def apply_role_post_chef_call(old_role, role, all_nodes)
+    all_nodes.each do |n|
+      node = NodeObject.find_node_by_name n
+      if node.role? "updater-upgrade"
+        # Put together storage role name
+        role_name = NodeObject.make_role_name(node.name)
+        role  = RoleObject.find_role_by_name(role_name+"-upgrade_storage")
+        role.name = role_name
+        node.delete_from_run_list("updater-upgrade")
+#        priority = runlist_priority_map[item] || local_chef_order
+        priority = 1
+        # FIXME: Not sure if this works, and if it does, find solution for priority
+        node.add_to_run_list(role.name, priority)
```

Comment:
Shouldn't this be enough?

    node.delete_from_run_list("updater-upgrade")
    node.add_to_run_list(make_role_name(node.name), priority)